### PR TITLE
File Reorg for queries and using object-spread

### PR DESCRIPTION
**Pull request by bsoe003**
* Reorganized file structure for queries.
  * `querys` is not grammatically correct, so changed to `queries`
  * each query exist as a `.gql` file.
  * `queries/index.js` will create a function based on reading each `.gql` file.
  * It is recommended to use GraphQL plugins for your editor to view syntax highlighting.
  * This enforces SRP and DRY.
  * **NOTE**: PLEASE TEST THIS FOR WEB MODULE. (new change uses `fs` which may not work outside of node)
* Change `Object.assign` to [object-rest-spread](https://github.com/tc39/proposal-object-rest-spread)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -64,6 +65,7 @@
     "babel-core": "6.26.3",
     "babel-loader": "7.1.4",
     "babel-plugin-transform-class-properties": "6.24.1",
+    "babel-plugin-transform-object-rest-spread": "^6.26.0",
```
Comment (bote795) [inline package.json]: can u set the version

File: webpack.config.js [hunk 2]
```diff
@@ -20,10 +20,12 @@ const web = {
       }
     ]
   },
+  node: {
+    fs: 'empty'
```
Comment (bote795) [inline webpack.config.js]: what does this do?

Comment (bsoe003) [reply webpack.config.js]: Since web doesn't know what `fs` is, webpack has to set it empty to build

File: package.json (hunk 1 above)
Comment (bsoe003) [reply package.json]: done

File: lib/queries/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable no-param-reassign, global-require, import/no-dynamic-require */
```
Comment (bsoe003) [inline lib/queries/index.js]: don't need this anymore.

**Comment by bote795** [comment]
@bsoe003 doesn't work on web, need to fix b4 merge

**Comment by bote795** [comment]
kept searching but doesn't seem there is a way to do it for web if we use `fs` so I don't think we should re-write using `fs`. 

- this is the doc for  [webpack node replacements](https://github.com/webpack/node-libs-browser)
- There is the pr to add it but not merged yet [pr](https://github.com/webpack/node-libs-browser/pull/64)

**Comment by bsoe003** [comment]
for some reason, my laptop keeps on downgrading my yarn. so I keep getting `cross-fetch` at `2.0.0`, so I can't manually test them

**Comment by bote795** [comment]
that is weird that it keeps downgrading your yarn :o

File: package.json [hunk 4]
```diff
@@ -35,7 +35,10 @@
     ],
     "plugins": [
       "transform-class-properties",
-      "transform-object-rest-spread"
+      "transform-object-rest-spread",
+      ["babel-plugin-inline-import", {
+        "extensions": [".json", ".gql"]
+      }]
```
Comment (bsoe003) [inline package.json]: need this. otherwise, mocha wouldn't know how to read `.gql` files

File: lib/queries/index.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import getUserListById from './getUserListById.gql';
+import removeMedia from './removeMedia.gql';
+import searchAnime from './searchAnime.gql';
+import updateMedia from './updateMedia.gql';
+
+const ops = { getUserListById, removeMedia, searchAnime, updateMedia };
+
+// transform each ops as a function that requests query
+Object.keys(ops).forEach(key => {
```
Comment (bote795) [inline lib/queries/index.js]: Cant we do this with reduce as well?

Comment (bsoe003) [reply lib/queries/index.js]: That would mean system would have to internally create an extra array, right?

Comment (bote795) [reply lib/queries/index.js]: True, btw do u want me to press merge or u wanna merge

Comment (bsoe003) [reply lib/queries/index.js]: I can't merge

Comment (bote795) [reply lib/queries/index.js]: gg lol will do that

**Pull request merged**